services/memory_storage/memory_storage.cpp

```diff
@@ -256,19 +274,82 @@ namespace services {
                   session.data());
             return;
         }
+        auto collection_names = logical_plan->collection_dependencies();
+        context_storage_t collection_dependencies;
+        for (const auto& name : collection_names) {
+            if (!check_collection_(session, name)) {
+                trace(log_,
+                      "memory_storage_t:execute_plan_ collection not found {}, sesion: {}",
+                      name.to_string(),
+                      session.data());
+                return;
+            }
+            collection_dependencies.emplace(name, collections_.at(name).context);
+        }
         sessions_.emplace(session, session_t{logical_plan, current_message()->sender(), 1});
-        actor_zeta::send(collections_.at(logical_plan->collection_full()),
+        auto plan =
+            collection::planner::create_plan(collection_dependencies, logical_plan, components::ql::limit_t::unlimit());
+
+        if (!plan) {
+            execute_plan_finish_(
+                session,
+                make_cursor(default_resource(), error_code_t::create_phisical_plan_error, "invalid query plan"));
+            return;
+        }
+
+        std::stack<collection::operators::operator_ptr> look_up;
```

Comment:
So here you traverse operation tree and put operation into sub plan from top->right->left order?
Could you move this logic into separate member? 
`auto plans_to_execute = traverse_plan(....);`